### DISCUSSION TITLE
Get current file path in lua filter

### DISCUSSION THREAD

**Discussion by jmbuhr** [quarto-dev/quarto-cli#1000] [Q&A]
I am working on a lua filter to add a "edit file locally" button to complement the "Edit this page on GitHub" button.
Is there a way to get the current file path from within a lua filter?
There is `PANDOC_STATE.input_files`, but because it is passed through quarto first, it will be something like `/tmp/quarto-sessioncb8bd231/3c9b23fe/quarto-input4c981533.md`.
`PANDOC_STATE.output_file` is as close as I got, since `filename.html` can easily be replaced with `filename.md`, but this still lacks the full filepath.

**Comment by jmbuhr** [comment]
This is how I ended up solving it: In `_quarto.yml` include something like
```
format:
  html:
    include-after-body: _filepath.html
```
where `_filepath.html` contains:
```
<script>
let basepath = '</local/path/to/quarto-project'
let path = window.location.pathname
let link = document.createElement('a')
if (path == '/wiki/') {
  path = path + 'index.html'
}
let url = 'vscode://file' + basepath + path.replace(/\/wiki/, '').replace(/\.html$/, '.md')
let text = document.createTextNode("Edit in vs code")
let paragraph = document.createElement('p')
link.appendChild(text);
link.setAttribute('href', url)
paragraph.appendChild(link)

let link2 = document.createElement('a')
let url2 = 'nvim://' + basepath + path.replace(/\/wiki/, '').replace(/\.html$/, '.md')
let text2 = document.createTextNode("Edit in neovim")
let paragraph2 = document.createElement('p')

link2.appendChild(text2);
link2.setAttribute('href', url2)
paragraph.appendChild(link2)


let firstH1 = document.getElementsByTagName('h1')[1]
firstH1.parentNode.insertBefore(paragraph, firstH1.nextSibling);
</script>

<style>
h1 + p {
    display: flex;
    flex-direction: column;
}
</style>
```
Which gives me these two links on top of each page:
![image](https://user-images.githubusercontent.com/17450586/203740999-74076ff6-5216-4744-95f3-8af31371c905.png)

**Reply by jmbuhr** [reply]
Note: VS code automatically registers itself as a url-handler. For neovim this needs to happen manually. E.g. by appending `x-scheme-handler/nvim=nvim-url.desktop` to `/usr/share/applications/defaults.list` or `~/.local/share/applications/defaults.list` where `nvim-url.desktop` is at `~/.local/share/applications/` and contains:
```
[Desktop Entry]
Version=1.1
Type=Application
Name=Neovim
GenericName=Text Editor
Comment=Edit text files
Icon=nvim
Exec=open-in-nvim.sh %u
MimeType=x-scheme-handler/nvim
Categories=TextEditor;Utility;
Keywords=Text;editor;
```
and `open-in-nvim.sh` has
```
#!/bin/env bash
kitty nvim $(echo $1 | sed 's/nvim:\/\///')
```
(the kitty part is optional, that's just the terminal I use)

**Comment by adivekar-utexas** [comment]
For me the "PWD" argument had useful information to get the dir when using Preview mode on VSCode

**Comment by cderv** [comment]
For reference, 

I believe this is related to this past discussion
- https://github.com/quarto-dev/quarto-cli/issues/2249

and new one where a workaround through an extension is shared
- https://github.com/quarto-dev/quarto-cli/issues/4770
